makefiles/build.mk

```diff
@@ -1,4 +1,18 @@
+SHELL := /bin/bash
+goreleaser_version=v1.24.0
+
+goreleaser-install:
+	@echo "--> Checking if goreleaser $(goreleaser_version) is installed"
+	@if [ "$$(goreleaser --version 2> /dev/null | grep GitVersion | awk '{ print $$2 }')" != "$(goreleaser_version)" ]; then \
+		echo "--> Installing goreleaser $(goreleaser_version)"; \
+		go install github.com/goreleaser/goreleaser@$(goreleaser_version); \
+	else \
+		echo "--> goreleaser $(goreleaser_version) is already installed"; \
+	fi
+
 build:
-	go build -o $(PKGNAME) -v ./cmd/$(PKGNAME)
-	
-.PHONY: build
\ No newline at end of file
+	@$(MAKE) goreleaser-install
+	goreleaser build --clean --snapshot
+	@echo "--> Build binary is available in the ./dist directory"
+
+.PHONY: build
```

Comment:
You need to add `goreleaser-install` to phony otherwise it will think it's associated to a file